### PR TITLE
MRTK - Near menu

### PR DESCRIPTION
**Pull request by CraigFeldspar**
+ Added `NearMenu` 3D control
+ Added `DefaultBehavior`, a behavior that will be common to several 3D controls, orchestrating `SixDoFDragBehavior`, `FollowBehavior`, and in the near future, `SurfaceMagnetismBehavior`
+ `SixDoFDragBehavior` can now specify an ancestor to drag instead of the attached mesh. It is used to make the back plate behave as an anchor to drag the whole control.

### CODE REVIEW

File: src/Misc/smoothingTools.ts
Comment (RaananW) [inline src/Misc/smoothingTools.ts:45]: Feels like those belong in the Math.Vector package. any reason you decided to offload them to a different file?

**Review by RaananW — COMMENTED** [review]
There are a few linting issues, I will gladly approve after it's all resolved.

A question regarding the .png file - I noticed you commited it to the assets lib, any reason why you commit it here as well?

File: src/Misc/smoothingTools.ts
Comment (CraigFeldspar) [reply src/Misc/smoothingTools.ts:45]: I thought at the moment it was a bit too specific to my case to belong to the Vector lib, so I added a helper in the model of `PivotTools`. But yeah  it can also belong to Math.vector

**Comment by CraigFeldspar** [comment]
> There are a few linting issues, I will gladly approve after it's all resolved.
> 
> A question regarding the .png file - I noticed you commited it to the assets lib, any reason why you commit it here as well?

It was just temporary waiting for the PR in the Assets repo to be merged. I'll remove it

File: dist/preview release/what's new.md
Comment (RaananW) [inline dist/preview release/what's new.md:1]: Would you be able to revert the changes to this file and make sure only your line changes?

**Review by RaananW — COMMENTED** [review]
Visualization test failed, want to check what happened there?

**Review by RaananW — APPROVED** [review]

File: gui/src/3D/controls/nearMenu.ts
Comment (Popov72) [inline gui/src/3D/controls/nearMenu.ts:54]: Should the user be able to update those colors (see below too)?

File: gui/src/3D/controls/nearMenu.ts
Comment (Popov72) [inline gui/src/3D/controls/nearMenu.ts:98]: Same question than above.

File: gui/src/3D/controls/nearMenu.ts
Comment (Popov72) [inline gui/src/3D/controls/nearMenu.ts:116]: nit: could be `const`

File: gui/src/3D/controls/nearMenu.ts
Comment (Popov72) [inline gui/src/3D/controls/nearMenu.ts:158]: nit: missing doc

File: gui/src/3D/controls/nearMenu.ts
Comment (Popov72) [inline gui/src/3D/controls/nearMenu.ts:197]: nit: missing doc

File: src/Behaviors/Meshes/sixDofDragBehavior.ts
Comment (Popov72) [inline src/Behaviors/Meshes/sixDofDragBehavior.ts:170]: nit: both could be `const`

**Review by Popov72 — COMMENTED** [review]
LGTM!

File: gui/src/3D/controls/nearMenu.ts
Comment (CraigFeldspar) [reply gui/src/3D/controls/nearMenu.ts:54]: This is really temp code as a new shader is coming to replace the FluentMaterial. There is also the same hard coded color in the GUI slate.

Comment (CraigFeldspar) [reply gui/src/3D/controls/nearMenu.ts:54]: https://github.com/BabylonJS/Babylon.js/pull/10397 <- waiting for this PR

File: gui/src/3D/controls/nearMenu.ts
Comment (CraigFeldspar) [reply gui/src/3D/controls/nearMenu.ts:98]: This too will be replaced :)

**Comment by RaananW** [comment]
Validation tests are failing

**Pull request merged**
